src/maison/utils.py

```diff
@@ -109,3 +109,43 @@ def _collect_configs(
             sources.append(IniSource(**source_kwargs))
 
     return sources
+
+
+def deep_merge(destination: Dict[Any, Any], source: Dict[Any, Any]) -> Dict[Any, Any]:
+    """Recursively updates the destination dictionary.
+
+    Args:
+        destination: A dictionary to be merged into. This will be updated in place.
+        source: The dictionary supplying data
+
+    Returns:
+        The updated destination dictionary.
+
+    Raises:
+        RuntimeError: A dict cannot be merged on top of a non-dict.
+          For example, the following would fail:
+          `deep_merge({"foo": "bar"}, {"foo": {"baz": "qux"}})`
```

Comment:
some indentation issues here

Reply:
I'm not sure how to fix this. I can't get flake8 to pass no matter what I do.

Reply:
looks like these two lines just need two more spaces of indentation?

Reply:
That does the trick! Done in https://github.com/dbatten5/maison/pull/89/commits/510faa78f00ed84851ddb7e25929ed3c360c25f4.

Reply:
great! one final hurdle is the coverage which i think could be sorted by adding a test to cover that `RuntimeError`. sorry i know this ci is quite strict